### PR TITLE
Fix call hierarchy item serialization

### LINKED ISSUE #37061: Add container information to `CallHierarchyItem`
From https://github.com/microsoft/vscode/issues/89616

## Search Terms
- call hierarchy
- tsserver

## Problem
Currently using the call hierarchy api, it can be difficult to browse through long list of callers and understand where those callers are:

Screen Shot 2020-02-26 at 3 12 17 PM

## Suggestion
Add optional information about the container on the `CallHierarchyItem` type. For example, the `NavTooItem` has:

```ts
containerName?: string;
containerKind?: ScriptElementKind;
```

We should consider something similar for `CallHierarchyItem`

/cc @jrieken

### PR DESCRIPTION
**Pull request by andrewbranch**
Fixes #37061

### CODE REVIEW

File: src/server/session.ts [hunk 1]
```diff
@@ -2408,6 +2408,7 @@ namespace ts.server {
                 kind: item.kind,
                 kindModifiers: item.kindModifiers,
                 file: item.file,
+                containerName: item.containerName,
```
Comment (andrewbranch) [inline src/server/session.ts:2411]: This line is the implementation fix. Other changes are fixes to test infrastructure that was never used before now.

**Comment by andrewbranch** [comment]
@typescript-bot cherry-pick this to release-4.0

**Comment by andrewbranch** [comment]
I’m guessing we want to get this in our VS Code servicing release. @DanielRosenwasser I’ve cherry picked this PR but should we wait to merge it until after another patch release?

**Comment by mjbvz** [comment]
Yes that would be great. We're in our endgame this week for VS Code so I don't think this will make it into VS Code 1.49.

We'd normally pick up the next TS servicing release by the end of September but can pick one up anytime for testing in insiders

**Pull request merged**